scala-thoughts.md

```diff
@@ -0,0 +1,14 @@
+This was my first time programming in Scala and it was definitely quite challenging. What was most challenging was finding the 
+Scala-y way of doign things. It was easy to resort to the imperative style of programming when I was unsure of how to approach
+a problem. Once I realized the pattern-matching capabilities of Scala, I ended up using it in a lot of the problems. The 
+functional style of programming gave me flashbacks to CS52 and SML. In terms of documentation, I found most of the resources 
+were a little too long-winded and not example-oriented enough, and it was annoying to have to frequently look at documentation
+but that's something that is part of the process of learning a new language. 

```

Comment:
I also did not find the documentation example oriented enough.  I found some nice stack overflow posts when I just googled the actual problem I was having/thing I wanted to do.